src/app.ts

```diff
@@ -1,84 +1,143 @@
+import type http from 'http'
+
+import { diContainer, fastifyAwilixPlugin } from '@fastify/awilix'
+import {
+  bugsnagPlugin,
+  getRequestIdFastifyAppConfig,
+  metricsPlugin,
+  publicHealthcheckPlugin,
+} from '@lokalise/fastify-extras'
 import fastify from 'fastify'
+import customHealthCheck from 'fastify-custom-healthcheck'
+import fastifyGracefulShutdown from 'fastify-graceful-shutdown'
 import type { ZodTypeProvider } from 'fastify-type-provider-zod'
 import { serializerCompiler, validatorCompiler } from 'fastify-type-provider-zod'
+import type pino from 'pino'
 
-import config from './config'
+import { getConfig, isDevelopment } from './infrastructure/config'
+import { registerDependencies } from './infrastructure/diConfig'
 import { errorHandler } from './infrastructure/errors/errorHandler'
-import integrationConfigPlugin from './plugins/integrationConfigPlugin'
-import routeDefinitions from './routes'
+import { resolveGlobalErrorLogObject } from './infrastructure/errors/globalErrorHandler'
+import { resolveLoggerConfiguration } from './infrastructure/logger'
+import { routeDefinitions } from './modules/routes'
+import { healthcheckPlugin } from './plugins/healthcheckPlugin'
+import { integrationConfigPlugin } from './plugins/integrationConfigPlugin'
 
+const GRACEFUL_SHUTDOWN_TIMEOUT_IN_MSECS = 10000
 const API_VERSION = '2.1.1'
 
 const getMajorApiVersion = (): string => {
   return parseInt(API_VERSION).toString()
 }
 
-const getApp = async () => {
-  const app = fastify({
-    logger:
-      config.app.env === 'development'
-        ? {
-            transport: {
-              target: 'pino-pretty',
-              options: {
-                colorize: true,
-                quietReqLogger: true,
-                crlf: true,
-                levelFirst: true,
-                singleLine: true,
-                messageFormat: '[{context}] {msg}',
-                errorLikeObjectKeys: [],
-                translateTime: 'SYS:standard',
-                ignore: 'hostname,pid',
-              },
-            },
-          }
-        : true,
+export function getPrefix() {
+  return `/v${getMajorApiVersion()}`
+}
+
+export type ConfigOverrides = {
+  enableMetrics?: boolean
+}
+
+export async function getApp(configOverrides: ConfigOverrides = {}) {
+  const config = getConfig()
+  const appConfig = config.app
+  const loggerConfig = resolveLoggerConfiguration(appConfig)
+  const enableRequestLogging = ['debug', 'trace'].includes(appConfig.logLevel)
```

Comment:
🟢 I suggest we add `['debug', 'trace']` in a constant

Reply:
fixed